rtt/base/DataObjectLockFree.hpp

```diff
@@ -198,34 +237,77 @@ namespace RTT
              * not increment the counter on write_ptr+1). Hence, no
              * locking is needed.
              */
+
+            if (!initialized) {
+                types::TypeInfo *ti = types::Types()->getTypeInfo<DataType>();
```

Comment:
You are right that not having set a data sample is not an error for many types and the error message should be degraded to warning or info level. On the other side I never saw this triggering for the normal usage of the lock-free data object in port connections, at least not for local connections. The OutputPort always sets a sample on the connection in [OutputPort.hpp:122](https://github.com/meyerj/rtt/blob/8d7c478d605a95271acefaa753703dd3fa548056/rtt/OutputPort.hpp#L122). I would have to verify if this is also the case for remote connections and/or streams.

Did you actually see this message while testing this branch with Rock?